### PR TITLE
chore: update to onchainkit@0.5.1, viem@2, wagmi@2

### PR DESCRIPTION
**Pull request by eragon512**
**What changed? Why?**

**Notes to reviewers**

**How has it been tested?**
Manual

### CODE REVIEW

**Review by Sneh1999 — COMMENTED** [review]
@eragon512 Rainbow Kit has released support for wagmi 2.0 in beta version. Maybe we can use that?

File: template/src/pageComponents/buy-me-coffee/steps/BuyCoffeeFormStep/FormBuyCoffee.tsx [hunk 1]
```diff
@@ -43,64 +43,90 @@ function FormBuyCoffee({
...
+    status: statusBuyMeACoffee,
+    error: errorBuyMeACoffee,
+  } = useWriteContract();
+
+  const { status: transactionStatus } = useWaitForTransactionReceipt({
+    hash: dataBuyMeACoffee,
+    query: {
+      enabled: !!dataBuyMeACoffee,
+    },
+    // onSuccess() {
```
Comment (Sneh1999) [inline template/src/pageComponents/buy-me-coffee/steps/BuyCoffeeFormStep/FormBuyCoffee.tsx:73]: Do we need to handle these?

File: template/src/pageComponents/mint/steps/StartMintStep.tsx [hunk 2]
```diff
@@ -12,30 +12,36 @@ type StartMintProps = {
...
+      enabled: onCorrectNetwork,
+    },
   });
 
   // A future enhancement would be to use the `isLoading` and `isSuccess`
   // properties returned by `useContractWrite` to indicate transaction
   // status in the UI.
-  const { write: performMint } = useContractWrite({
-    ...config,
-    onSuccess() {
```
Comment (Sneh1999) [inline template/src/pageComponents/mint/steps/StartMintStep.tsx:35]: I believe we should handle the `onSuccess` and `onError` state here as well

**Review by Sneh1999 — COMMENTED** [review]
LGTM! Added a few comments

File: template/package.json [hunk 3]
```diff
@@ -18,13 +18,14 @@
...
   "dependencies": {
-    "@coinbase/onchainkit": "0.4.1",
+    "@coinbase/onchainkit": "0.5.1",
     "@coinbase/wallet-sdk": "^3.9.1",
     "@farcaster/hub-nodejs": "^0.10.21",
     "@radix-ui/react-dropdown-menu": "^2.0.6",
     "@radix-ui/react-icons": "^1.3.0",
     "@radix-ui/react-navigation-menu": "^1.1.4",
-    "@rainbow-me/rainbowkit": "1.3.3",
+    "@rainbow-me/rainbowkit": "2.0.0-beta.6",
```
Comment (Zizzamia) [inline template/package.json:27]: LFG

File: template/src/pageComponents/mint/steps/StartMintStep.tsx (hunk 2 above)
Comment (eragon512) [reply template/src/pageComponents/mint/steps/StartMintStep.tsx:35]: `onSuccess` and `onError` callbacks have been removed in wagmi@2

File: template/src/pageComponents/buy-me-coffee/steps/BuyCoffeeFormStep/FormBuyCoffee.tsx (hunk 1 above)
Comment (eragon512) [reply template/src/pageComponents/buy-me-coffee/steps/BuyCoffeeFormStep/FormBuyCoffee.tsx]: onSuccess and onError callbacks have been removed in wagmi@2

File: template/src/pageComponents/mint/steps/StartMintStep.tsx (hunk 2 above)
Comment (Sneh1999) [reply template/src/pageComponents/mint/steps/StartMintStep.tsx:35]: @eragon512 you can follow this pattern, I did [here](https://github.com/coinbase/build-onchain-apps/pull/338/files#diff-b1bd0f0d1a27dc6af7559a99bb29dcfdc796a1467a1b877fa6df93da685d6819R58)

Let me know if this makes sense!

Comment (eragon512) [reply template/src/pageComponents/mint/steps/StartMintStep.tsx:35]: @Sneh1999 updated with a similar pattern - need some help with verifying the smart contract behaviour though

**Pull request merged**